### PR TITLE
[move-prover] fix error filtering in boogie-wrapper

### PR DESCRIPTION
**Pull request by mengxu-fb**
Currently the prover collects three types of errors from Boogie:
- verification errors
- inconclusive errors
- inconsistency errors

In addition, in the collector for verification errors, there is a check
that if an error message does not belong to other known categories of
errors, it will mark the error message as an unknown boogie error.

Unfortunately the filter is not working now (maybe because Boogie has
changed the output format), leading to error messages being marked as
unknown (while they should be marked as inconclusive).

This commit re-organized the error parsing logic and uses the REGEX
patterns consistently for both early filtering as well as actual error
parsing.

## Motivation

Fix the daily CI

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

```
cd language/move-prover
mvp --dependency=../move-stdlib/modules --dependency=../diem-framework/modules --check-inconsistency --stable-test-output ../diem-framework/modules/DiemId.move
```

Expect no error.

Before the fix, expect the following error:

```
error: unexpected boogie output: `output.bpl(13461,85): Verification of '$DiemId_remove_diem_id_domain$verify_inconsistency' timed out after 40 seconds ..`

   ┌── <unknown>:1:1 ───
   │
 1 │ <unknown>
   │ ^^^^^^^^^
   │
```

The reason is that error message
```$DiemId_remove_diem_id_domain$verify_inconsistency' timed out after 40 seconds ..```
is not recognized by the verification error parser and hence, this error message is marked as an unknown boogie error.

Although later this message is indeed picked up by the inclusive error parser and subsequently filtered out because of the presence of `$verify_inconsistency`, the message is not removed from the unknown boogie error list.

### CODE REVIEW

**Review by wrwg — DISMISSED** [review]

**Comment by mengxu-fb** [comment]
Removed the canary test

**Review by junkil-park — APPROVED** [review]
Thank you for the fix!

File: language/move-prover/boogie-backend/src/boogie_wrapper.rs [hunk 1]
```diff
@@ -355,33 +367,26 @@ impl<'env> BoogieWrapper<'env> {
...
+                && !INCONSISTENCY_DIAG_STARTS.is_match(inbetween)
+            {
                 // This is unexpected text and we report it as an internal error
                 errors.push(BoogieError {
                     kind: BoogieErrorKind::Internal,
                     loc: self.env.unknown_loc(),
                     message: format!(
                         "unexpected boogie output: `{} ..`",
-                        &inbetween[0..inbetween.len().min(70)]
+                        &inbetween[0..inbetween.len().min(120)]
```
[the remark this answers is not in the document]
Comment (mengxu-fb) [reply language/move-prover/boogie-backend/src/boogie_wrapper.rs]: Re-org the logic is enough, this change is reverted.

**Comment by mengxu-fb** [comment]
Unexpected failure of `crypto-unit-test` which this PR has nothing to do with. Retry landing

/land

**Pull request merged**